FirebaseAuth/Tests/Unit/AuthNotificationManagerTests.swift

```diff
@@ -51,7 +51,7 @@
         storage: FakeAuthKeychainStorage()
       )
       appCredentialManager = AuthAppCredentialManager(withKeychain: fakeKeychain)
-      let application = FakeApplication()
+      let application = UIApplication.shared
```

Comment:
Hmm I think it's reasonable to want to have a fake here at some point, maybe add a TODO for us to revisit at some point? I'm not sure why the above code in the AuthNotificationManager file didn't work as expected (passing in the `Application` which `UIApplication` conforms to)

Reply:
The fake doesn't work because of line 124 in [AuthNotificationManager.swift](https://github.com/firebase/firebase-ios-sdk/pull/11629/files#diff-f1b4b4d175c9f96bdc92c940bff9ce5cd09d4e221d1eec17e3257706c12f94f3). There will be a type mismatch on the first parameter.